civismlext/test/test_preprocessing.py

```diff
@@ -162,6 +163,27 @@ def test_sklearn_api():
             "to a dataframe.") in str(e.value)
 
 
+def test_flag_nulls(data_raw):
+    expander = DataFrameETL()
+    drop_cols = ['col1', 'col2']
+    drop_cols_2 = ['col1', 'col2', 'nantastic']
+    assert expander._flag_nulls(data_raw, []) == []
+    assert expander._flag_nulls(data_raw, drop_cols) == drop_cols
+
+    # add a col of all nans
+    data_raw['nantastic'] = pd.Series([np.NaN] * 3)
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        assert expander._flag_nulls(data_raw, drop_cols) == drop_cols_2
+        assert len(w) == 1
+        assert issubclass(w[-1].category, UserWarning)
+
+        # check that we don't add the col if it's already being dropped
+        assert expander._flag_nulls(data_raw, drop_cols_2) == drop_cols_2
+        assert len(w) == 1
+        assert issubclass(w[-1].category, UserWarning)
```

Comment:
Small thing, but I'd put this check first, so that you can have `assert len(w) == 0`. It's mildly confusing to assert one warning in a test which shouldn't warn.